core/src/main/java/io/ably/lib/realtime/AblyRealtimeBase.java

```diff
@@ -183,6 +208,21 @@ public void suspendAll(ErrorInfo error, boolean notifyStateChange) {
             }
         }
 
+        /**
+         * By spec RTN15c6, RTN15c7
+         */
+        @Override
+        public void reAttach() {
+            for (Map.Entry<String, RealtimeChannelBase> entry : map.entrySet()) {
+                RealtimeChannelBase channel = entry.getValue();
+                if (channel.state == ChannelState.attaching || channel.state == ChannelState.attached || channel.state == ChannelState.suspended) {
```

Comment:
Can you check if there is a method like `transition` exist in ably-java? I had done the same thing in ably-cocoa and Marat pointed out that this should be happening on its own transition method 🤔

Reply:
There is no such method, only validateTransition but it is abstract method that does all kinds of stuff across SDK. I cannot find anything similar to transition in java SDK.